### ISSUE TITLE
Blog: author profile list file

### ISSUE THREAD

**Issue by Josh-Cena** [facebook/docusaurus#5363]
## 🚀 Feature

A separate author profile list file to prevent duplicating author information in every markdown front matter.

### Have you read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

### Has this been requested on [Canny](https://docusaurus.io/feedback)?

Actually yes. https://docusaurus.io/feedback/p/add-unified-authors-data-file-to-reduce-header-attributes-for-blog-posts

## Motivation

From the canny request:

> In Jekyll, there's a `authors.yml` file that stores all authors' profile which includes social media links, author title and avatar url, so that we could just put only author attribute in blog header without author_title, author_url, author_image_url... in every post.

## API Design

The [Jekyll design](https://jekyllrb.com/docs/datafiles/#example-list-of-members), just for reference. We can have a `/blog/data/authors.yml` (JSON can work as well, of course), in which we put all the authors:

```yml
- Author 1
  name: ...
  url: ...
  image_url: ....
  title: ...
- Author 2
  name: ...
  url: ...
  image_url: ....
  title: ...
```

And then use them with a new `author_key: Author 1` front matter in a blog post. User can still override values in the YML within the post's front matter.

## Have you tried building it?

Not yet, would be willing to work on it

**Comment by slorber** [comment]
Yes we should definitively support that, as long as we can stay retrocompatible with existing frontmatter.

My blog Gatsby theme (Novela) also has such feature: https://github.com/slorber/sebastienlorber.com/tree/master/content/authors

**Comment by Josh-Cena** [comment]
@shakyapeiris Yes, although you can't be exempted from front matter entirely though because you have to use an `author_key` front matter to reference that external data

**Comment by shakyapeiris** [comment]
Ok Thanks

**Comment by slorber** [comment]
We also have a need for blog posts with multiple authors.

https://github.com/facebook/docusaurus/discussions/4527

So we'd likely use a frontmatter like `author_ids: [author1, author2]` to support that.

Not sure it's worth supporting multiple authors with existing frontmatter attributes. It would probably require a breaking change so that the existing frontmatter can support the list, so we may do this but later.

Note: author image URL can use a co-located asset in the folder's blog post now. Not sure we need this feature for "global blog authors" though

**Comment by Josh-Cena** [comment]
> Note: author image URL can use a co-located asset in the folder's blog post now. Not sure we need this feature for "global blog authors" though

I tried a very rough implementation and I noticed it's hard to co-locate assets in the global file. I guess we can only support using URLs to reference images in global files, but allow overriding this with a collocated image in front matter?

We can certainly make `author_key` accept multiple IDs

**Comment by Josh-Cena** [comment]
There has also been requests for author pages:

- https://docusaurus.io/feature-requests/p/add-all-authors-and-posts-by-author-to-ease-list-posts-by-author-in-blog
- https://docusaurus.io/feature-requests/p/blog-author-pages
- https://docusaurus.io/feature-requests/p/add-athors-page-like-tags-page (this one's description seems to be requesting multi-author blog posts though)

After we have unique author keys this feature may be easier to implement?
